### PR TITLE
c: Use WARN_UNUSED for ByteExtract* functions

### PR DESCRIPTION
**Pull request by hsadia538**
- [X] I have read the contributing guide lines at https://redmine.openinfosecfoundation.org/projects/suricata/wiki/Contributing
- [X] I have signed the Open Information Security Foundation contribution agreement at https://suricata.io/about/contribution-agreement/
- [ ] I have updated the user guide (in doc/userguide/) to reflect the changes made (if applicable)

Link to [redmine](https://redmine.openinfosecfoundation.org/projects/suricata/issues/3658) ticket:

**Draft PR for discussion and feedback**

Describe changes:
- Adding WARN_UNUSED for all ByteExtract functions in `util-byte.h`
- Fix warnings generated in code due to the macro

### CODE REVIEW

File: src/app-layer-enip-common.c [hunk 1]
```diff
@@ -91,10 +91,12 @@ static int ENIPExtractUint32(uint32_t *res, const uint8_t *input, uint16_t *offs
         return 0;
     }
 
-    ByteExtractUint32(res, BYTE_LITTLE_ENDIAN, sizeof(uint32_t),
-            (const uint8_t *) (input + *offset));
-    *offset += sizeof(uint32_t);
-    return 1;
+    if(ByteExtractUint32(res, BYTE_LITTLE_ENDIAN, sizeof(uint32_t),
```
Comment (hsadia538) [inline src/app-layer-enip-common.c:94]: The warnings are generated due to the output of this function being never actually used in the call.

The purpose of the caller function `ENIPExtractUint32` seems to get satisfied even without a return value being called and used from the `ByteExtractUint32` function.

The only thing that is making sense to me is adding an `if{}` like above and function returning success if bytes get extracted i.e `-1` is not returned.

Is this the right approach?

Comment (victorjulien) [reply src/app-layer-enip-common.c:94]: Doesn't look like it is. In other anomaly conditions, we return `0` (e.g. not enough input bytes), so if `ByteExtractUint32` fails we should probably do the same.

File: src/util-byte.h [hunk 2]
```diff
@@ -464,7 +464,7 @@ int WARN_UNUSED StringParseI16RangeCheck(
  * \param len Number of bytes to extract (23 max or 0 for unbounded)
  * \param str String to extract from
  *
- * \return n Number of bytes extracted on success
+ * \return n Number of bytes extracted on su./configccess
```
Comment (victorjulien) [inline src/util-byte.h:467]: accidental change by the looks of it

**Review by victorjulien — CHANGES_REQUESTED** [review]

File: src/util-byte.h (hunk 2 above)
Comment (hsadia538) [reply src/util-byte.h:467]: I think so, I will resolve this in next PR

File: src/app-layer-enip-common.c (hunk 1 above)
Comment (hsadia538) [reply src/app-layer-enip-common.c:94]: I did not quite understand I think. You are suggesting that here we should return 0 if `ByteExtractUint32 ` fails?

Comment (inashivb) [reply src/app-layer-enip-common.c:94]: Yes

**Comment by hsadia538** [comment]
Followed by: #8022

**Pull request closed without merging**